src/main/java/de/tum/in/www1/artemis/repository/AchievementRepository.java

```diff
@@ -0,0 +1,23 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.util.Set;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.Achievement;
+
+/**
+ * Spring Data JPA repository for the Achievement entity.
+ */
+@Repository
+public interface AchievementRepository extends JpaRepository<Achievement, Long> {
+
+    @Query("SELECT a FROM Achievement a WHERE a.course.id = :#{#courseId}")
+    Set<Achievement> getAllByCourseId(@Param("courseId") Long courseId);
+
+    @Query("SELECT a FROM Achievement a JOIN a.users u WHERE u.id = :#{#userId}")
+    Set<Achievement> getAllByUserId(@Param("userId") Long userId);
+}
```

Comment:
I think we need here also `getAllByUserIdAndCourseId` as it can be that students are registered for several courses.

Reply:
Yeah sure. 

My idea was to display a list of all achievements in a course in its dashboard with the achievements not yet received grayed out and the earned ones in color. So you would have to get all achievements for the course but without the user being able to see other students' information. 

But it doesn´t hurt to also have the possibility to get achievements by course id and user id.